craft_application/services/init.py

```diff
@@ -27,12 +27,21 @@
 from craft_application.errors import InitError
 from craft_application.git import GitError, GitRepo, is_repo, parse_describe
 
+from ..models.constraints import MESSAGE_INVALID_NAME, PROJECT_NAME_COMPILED_REGEX
 from . import base
 
 
 class InitService(base.AppService):
     """Service class for initializing a project."""
 
+    def validate_project_name(self, name: str) -> None:
+        """Validate that ``name`` is valid as a project name.
+
+        Raises an InitError if ``name`` is invalid.
+        """
+        if not PROJECT_NAME_COMPILED_REGEX.match(name):
```

Comment:
Wouldn't it be easier to just create `ProjectName(name)` and raise if `pydantic.ValidationError` occurs ?

Reply:
I started doing that but `ProjectName` is just a str, it needs to go through a Pydantic model for this
and then the code got too complicated and this ended up being simpler